app/controllers/collections_controller.rb

```diff
@@ -25,11 +24,13 @@ def new
   def edit; end
 
   def create
-    @collection = Collection.new(collection_params)
-    @collection.users << current_user
+    @collection = Collection.new(collection_params.merge(users: [current_user]))
     authorize @collection
+    create_with_task = @collection.tasks.any?
 
-    if @collection.save
+    if create_with_task
```

Comment:
I'd rather add an explanatory code comment here instead of making it a descriptive variable.
Benefit: A code comment can focus on the _WHY_. Why is the redirect behavior different if there are any tasks?